src/repr/src/proto.rs

```diff
@@ -36,6 +36,9 @@ pub enum TryFromProtoError {
     /// Indicates an `Option<U>` field in the `Proto$T` that should be set,
     /// but for some reason it is not. In practice this should never occur.
     MissingField(String),
+    /// Indicates that the serialized ShardId value failed to deserialize, according
+    /// to its custom deserialization logic.
+    InvalidShardId(String),
```

Comment:
this seems fine to me, @aalexandrov might have something to say

Reply:
I actually asked [about this on Slack](https://materializeinc.slack.com/archives/C02PPB50ZHS/p1652279290818419) and he was okay with adding more errors specific to Storage/etc (which is why I added a bunch of them!).

Reply:
very nice!